src/main/java/org/kuali/kfs/module/purap/document/PurchaseOrderDocument.java

```diff
@@ -777,16 +787,24 @@ protected String getCurrentRouteNodeName(WorkflowDocument wd) throws WorkflowExc
      * @throws WorkflowException
      */
     public void appSpecificRouteDocumentToUser(WorkflowDocument workflowDocument, String routePrincipalId, String annotation, String responsibility) throws WorkflowException {
-        boolean isActiveUser = this.isActiveUser(routePrincipalId);
-        if(!isActiveUser){
-            LOG.info("cannot send FYI to the inactive user: " + routePrincipalId + "; Annotation: " + annotation);
-            return;
-        }
         if (ObjectUtils.isNotNull(workflowDocument)) {
-            String annotationNote = (ObjectUtils.isNull(annotation)) ? "" : annotation;
-            String responsibilityNote = (ObjectUtils.isNull(responsibility)) ? "" : responsibility;
-            String currentNodeName = getCurrentRouteNodeName(workflowDocument);
-            workflowDocument.adHocToPrincipal( ActionRequestType.FYI, currentNodeName, annotationNote, routePrincipalId, responsibilityNote, true);
+            boolean isActiveUser = this.isActiveUser(routePrincipalId);
+            Map<String, String> permissionDetails = new HashMap<String, String>();
+            permissionDetails.put(KimConstants.AttributeConstants.DOCUMENT_TYPE_NAME, workflowDocument.getDocumentTypeName());
+            permissionDetails.put(KimConstants.AttributeConstants.ACTION_REQUEST_CD, KewApiConstants.ACTION_REQUEST_FYI_REQ);
+            boolean canReceiveAdHocRequest = KimApiServiceLocator.getPermissionService().isAuthorizedByTemplate(routePrincipalId, KewApiConstants.KEW_NAMESPACE, KewApiConstants.AD_HOC_REVIEW_PERMISSION, permissionDetails, new HashMap<String, String>());
```

Comment:
would it be worthwhile to clean this up by moving this to a function to support future testing?